pkgs/by-name/ms/msecli/package.nix

```
@@ -0,0 +1,65 @@
+{
+  autoPatchelfHook,
+  buildFHSEnv,
+  fetchurl,
+  lib,
+  stdenv,
+  zlib,
+}:
+
+let
+  pname = "msecli";
+  version = "10.01.012024.00";
+
+  src = fetchurl {
+    url = "https://www.micron.com/content/dam/micron/global/public/products/software/storage-executive-software/msecli/msecli-linux.run";
+    hash = "sha256-IszdD/9fAh+JA26bSR1roXSo8LDU/rf4CuRI3HjU1xc=";
+  };
+
+  buildEnv = buildFHSEnv { name = "msecli-buildEnv"; };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
+
+  buildInputs = [ zlib ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    cp "$src" ${src.name}
+    chmod +x ${src.name}
+
+    runHook postUnpack
+  '';
+
+  buildPhase = ''
+    runHook prebuild
+
+    # ignore the exit code as the installer
+    # fails at optional steps due to read only FHS
+    ${buildEnv}/bin/${buildEnv.name} -c "./${src.name} --mode unattended --prefix bin || true"
```

```suggestion
    ${buildEnv}/bin/${buildEnv.name} -c "./${src.name} --mode unattended --prefix $out || true"
```
Can we send this directly to $out and skip the installPhase?

---

I tried out the suggestion. Unfortunately the bubblewrap implementation of buildFHSEnv mounts the /nix as read only, and in doing so prevents the buildEnv from writing directly to $out. https://github.com/NixOS/nixpkgs/blob/245ce924200529111f41156ccd0689b646df6347/pkgs/build-support/build-fhsenv-bubblewrap/default.nix#L239

As a simple test I added 
```nix
${buildEnv}/bin/${buildEnv.name} -c "mkdir -p $out/bin"
```
before the installer command and it resulted in the following error during build:

```
mkdir: cannot create directory '/nix/store/vzd2vpkzim76i5w22bbiab9jgxyh2m7g-msecli-10.01.012024.00': Read-only file system
```

---

That's unfortunate. Leave it as is then.